runtime/doc/lua.txt

```diff
@@ -1056,6 +1056,8 @@ vim.fn.{func}({...})					*vim.fn*
         Note: vim.fn keys are generated lazily, thus `pairs(vim.fn)` only
         enumerates functions that were called at least once.
 
+        Note: The majority of functions cannot run in fast callbacks with some
+        undocumented exceptions which are allowed.
```

Comment:
once we get around to generating eval.txt, this will follow. https://github.com/neovim/neovim/issues/18311